be/src/formats/parquet/complex_column_reader.cpp

```diff
@@ -271,7 +271,17 @@ bool StructColumnReader::try_to_use_dict_filter(ExprContext* ctx, bool is_decode
     if (_child_readers[sub_field] == nullptr) {
         return false;
     }
-    return _child_readers[sub_field]->try_to_use_dict_filter(ctx, is_decode_needed, slotId, sub_field_path, layer + 1);
+    return _child_readers[sub_field]->can_use_dict_filter(ctx, is_decode_needed, slotId, sub_field_path, layer + 1);
+}
+
+bool StructColumnReader::try_to_use_dict_filter(ExprContext* ctx, bool is_decode_needed, const SlotId slotId,
+                                                const std::vector<std::string>& sub_field_path, const size_t& layer) {
+    if (this->can_use_dict_filter(ctx, is_decode_needed, slotId, sub_field_path, layer)) {
+        const std::string& sub_field = sub_field_path[layer];
```

Comment:
Hi @SHaaD94 , I see can_use_dict_filter is very similar with try_to_use_dict_filter, why we need to call it again?

Reply:
`can_use_dict_filter` does not change reader's state, `try_to_use_dict_filter` does - this is how the problem originated, we tried to enable for all expressions of slot even though it couldn't be used for some of them

Reply:
We are calling it again in `try` method to ensure it can be used, as an additional safety net, to not get crash of any sort